src/controllers/things_controller.js

```diff
@@ -232,17 +232,51 @@ ThingsController.put(
       return;
     }
     const value = request.body[propertyName];
-    AddonManager.setProperty(thingId, propertyName, value)
-      .then((updatedValue) => {
-        // Note: it's possible that updatedValue doesn't match value.
-        const result = {};
-        result[propertyName] = updatedValue;
-        response.status(200).json(result);
-      }).catch((error) => {
-        console.error('Error setting value for thingId:', thingId,
-                      'property:', propertyName,
-                      'value:', value);
-        response.status(500).send(error);
+    Things.getThingDescription(thingId, request.get('Host'), request.secure)
+      .then((thing) => {
+        if (!thing.properties.hasOwnProperty(propertyName)) {
+          response.status(404).end();
+          return;
+        }
+
+        if (thing.properties[propertyName].readOnly) {
+          response.status(400).end();
```

Comment:
I'd prefer if all of these error codes had an accompanying message so that a developer doesn't have to wonder why they're getting a 400